src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,50 +1,177 @@
+        int sizeList = addedList.length;
+        if (sizeList == 0) {
+            return false;
+        }
+        Node<T> before = last;
+
+        for (Object o : addedList) {
+            Node<T> newNode = new Node<>(before, (T) o, null);
+            if (before == null) {
+                first = newNode;
+            } else {
+                before.next = newNode;
+                before = newNode;
+            }
+        }
+
+        last = before;
+        size += sizeList;
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return findByIndex(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> setNode = findByIndex(index);
+        T oldValue = setNode.element;
+        setNode.element = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> removeNode = findByIndex(index);
+        return removeHelper(removeNode);
     }
 
     @Override
     public boolean remove(T object) {
-        return false;
+        Node<T> currentNode = first;
+        Node<T> removeNode = null;
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(currentNode.element, object)) {
+                removeNode = currentNode;
+                break;
+            }
+            currentNode = currentNode.next;
+        }
+
+        if (removeNode == null) {
+            return false;
+        }
+        removeHelper(removeNode);
+        return true;
+    }
+
+    private T removeHelper(Node<T> removeNode) {
+        Node<T> before = removeNode.prev;
+        Node<T> after = removeNode.next;
+        final T removeElement = removeNode.element;
+
+        if (before == null) {
+            first = after;
+        } else {
+            before.next = after;
+            removeNode.prev = null;
+        }
+
+        if (after == null) {
+            last = before;
+        } else {
+            after.prev = before;
+            removeNode.next = null;
+        }
+        size--;
+        return removeElement;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> findByIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException(EXCEPTION_MESSAGE);
+        }
+        if (index == 0) {
+            return first;
+        } else if (index == size - 1) {
+            return last;
+        }
+        int variableForCatchIndex = 0;
+        Node<T> findNode = first;
+
+        while (findNode.next != null) {
+            findNode = findNode.next;
+            if (index == ++variableForCatchIndex) {
+                return findNode;
+            }
+        }
+        throw new IndexOutOfBoundsException(EXCEPTION_MESSAGE);
```

Comment:
You dont need it here. You have alredy checked your index 
```suggestion
        
```

Reply:
done